driver-core/src/test/functional/com/mongodb/internal/connection/ServerHelper.java

```diff
@@ -62,15 +61,25 @@ public static void waitForLastRelease(final ServerAddress address, final Cluster
     }
 
     private static void checkPool(final ServerAddress address, final Cluster cluster) {
-        DefaultServer server = (DefaultServer) cluster.selectServer(new ServerAddressSelector(address))
-                .getServer();
-        DefaultConnectionPool connectionProvider = (DefaultConnectionPool) server.getConnectionPool();
-        ConcurrentPool<UsageTrackingInternalConnection> pool = connectionProvider.getPool();
+        ConcurrentPool<UsageTrackingInternalConnection> pool = connectionPool(
+                cluster.selectServer(new ServerAddressSelector(address)).getServer());
         if (pool.getInUseCount() > 0) {
             throw new IllegalStateException("Connection pool in use count is " + pool.getInUseCount());
         }
     }
 
+    private static ConcurrentPool<UsageTrackingInternalConnection> connectionPool(final Server server) {
+        ConnectionPool connectionPool;
+        if (server instanceof DefaultServer) {
+            connectionPool = ((DefaultServer) server).getConnectionPool();
+        } else if (server instanceof LoadBalancedServer) {
```

Comment:
All the changes in `ServerHelper` are done to support running `QueryBatchCursorFunctionalSpecification` and `AsyncQueryBatchCursorFunctionalSpecification` with load balancer.